crates/net/network-types/src/peers/mod.rs

```diff
@@ -35,6 +36,9 @@ pub struct Peer {
     /// Counts number of times the peer was backed off due to a severe
     /// [`BackoffKind`](crate::BackoffKind).
     pub severe_backoff_counter: u8,
+    /// Original [`TrustedPeer`] information for hostname-based peers, set only for local trusted
+    /// peers, `admin_addTrustedPeer` does not support hostname for node address
+    pub trusted_peer_info: Option<TrustedPeer>,
```

Comment:
need to think about this a bit,

but in any case we should box this here because this will be None most of the time